lib/bemxjst/tree.js

```diff
@@ -353,6 +353,10 @@ Tree.prototype.body = function() {
 
 Tree.prototype.match = function() {
   var children = new Array(arguments.length);
+
+  if (!arguments.length)
+    throw new Error('.match() must have argument');
```

Comment:
For the point point of view: you dont need test case for `assert(arguments.length)`